### DISCUSSION TITLE
More partition flexibility for the installer?

### DISCUSSION THREAD

**Discussion by purevw** [pop-os/pop#2570] [Q&A]
My typical Linux partition setup for openSUSE (and other Linux systems) is as follows:
100MB mounted as /boot/efi   format Fat32
500MB mounted as /boot  format ext4
30 GB mounted as / format ext4 (encrypted)
2GB mounted as swap (encrypted)
balance of drive mounted as /home format ext4 (encrypted)

I can format the petitions, but the installer does not allow the mount points. I set the 100MB partition as /boot/efi and am given output that it is not large enough. 100MB is more than large enough for /boot/efi but apparently the installer requires that /boot and /boot/efi be on the same partition. Is there a workaround to force the installer to accept separate partitions for /boot and /boot/efi?

**Comment by jacobgkau** [comment]
> 100MB is more than large enough for /boot/efi

It is not. Here is output showing my own `/boot/efi` subdirectory sizes (on a default-partition system):

```
121M	./EFI/Recovery-ED25-CE49
245M	./EFI/Pop_OS-92abb9cf-b62d-4c49-93c9-ef9a18a3abce
96K	./EFI/systemd
96K	./EFI/BOOT
4.0K	./EFI/Linux
366M	./EFI
16K	./loader/entries
28K	./loader
4.0K	./25d5909269b53db180df4b1060538eef
366M	.
```

Even if you don't have a recovery partition, the current and N-1 kernel/initrd versions take up more than twice 100M. Even just the most current initrd takes up more than 100M.

```
-rwx------ 1 root root  179 Aug 11 13:04 cmdline
-rwx------ 1 root root 113M Aug 11 13:04 initrd.img
-rwx------ 1 root root 111M Aug 11 13:04 initrd.img-previous
-rwx------ 1 root root  11M Aug 11 13:04 vmlinuz.efi
-rwx------ 1 root root  11M Aug 11 13:04 vmlinuz-previous.efi
```

There should be no problem having separate /boot and /boot/efi partitions (although separating /boot out from / seems a little pointless, but I assume you have your reasoning.) The EFI system partition needs to be large enough to fit the kernel and initrd, though.

**Reply by mindaugasvcs** [reply]
What makes you think you can place whole kernel files in EFI partition? I don't see such behaviour on Windows and other Linux distros

**Comment by purevw** [comment]
The Live USB is my recovery partition. No other is needed. My /boot/efi in openSUSE is 5MB, which is pretty much standard for an actual efi boot partition. The kernel and initrd reside within /boot and not within /boot/efi . 
`total 12
drwxr-xr-x 2 root root 4096 Jun 15 19:23 $RECYCLE.BIN
drwxr-xr-x 4 root root 4096 Jun 15 19:48 EFI
drwxr-xr-x 2 root root 4096 Jun 15 19:23 System Volume Information

./$RECYCLE.BIN:
total 4
-rwxr-xr-x 1 root root 129 Jun 15 19:23 desktop.ini

./EFI:
total 8
drwxr-xr-x 2 root root 4096 Jun 15 19:48 boot
drwxr-xr-x 2 root root 4096 Jun 15 19:48 opensuse

./EFI/boot:
total 1832
-rwxr-xr-x 1 root root 846240 Jun 15 19:48 MokManager.efi
-rwxr-xr-x 1 root root 934680 Jun 15 19:48 bootx64.efi
-rwxr-xr-x 1 root root  86352 Jun 15 19:48 fallback.efi

./EFI/opensuse:
total 3104
-rwxr-xr-x 1 root root  846240 Jun 17 16:19 MokManager.efi
-rwxr-xr-x 1 root root      58 Jun 17 16:19 boot.csv
-rwxr-xr-x 1 root root     120 Jun 17 16:19 grub.cfg
-rwxr-xr-x 1 root root 1239040 Jun 17 16:19 grub.efi
-rwxr-xr-x 1 root root  143360 Jun 17 16:19 grubx64.efi
-rwxr-xr-x 1 root root  934680 Jun 17 16:19 shim.efi

./System Volume Information:
total 8
-rwxr-xr-x 1 root root 76 Jun 15 19:23 IndexerVolumeGuid
-rwxr-xr-x 1 root root 12 Jun 15 19:23 WPSettings.dat
`
So I am assuming that typical Linux efi partitioning cannot be done?

**Reply by jacobgkau** [reply]
You did not reply to my post correctly, you started a new answer thread.

**Reply by jacobgkau** [reply]
You are using words like "standard" and "typical" to refer to your personal openSUSE configuration. There is nothing wrong with your openSUSE configuration, but Pop!_OS uses systemd-boot as a bootloader by default and places the kernel and initrd in the EFI system partition by default (this allows the rest of `/boot` to be encrypted along with `/` in the default partitioning setup; the ESP needs to be unencrypted since the initrd needs to be loaded before `/` can be decrypted.) The installer has checks to prevent a user from creating a partitioning setup that would not work with the default configuration.

It looks like you may be dual-booting with Windows and that is why you're attempting to use GRUB? UEFI specifications allow multiple EFI system partitions on a system, and UEFI firmware should allow you to choose your OS before reaching any OS bootloader; you have the option of simply giving Pop a larger EFI system partition instead of its own boot partition (resulting in the same amount of space being reserved), or creating a larger ESP and attempting to share it between multiple OS's.

The source code for the installer frontend is located at https://github.com/pop-os/installer and the backend is at https://github.com/pop-os/distinst if you wish to dig further into where and how the installer's restrictions are enforced.

**Reply by purevw** [reply]
Actually I used the word typical because it is typical for openSUSE, as is grub2. grub2 is the default bootloader. I do have a dual OS machine, but I select which boot device to use via BIOS. I do not dual boot via grub2 nor Windows. I do thank you for the information. It is much more than I understood previously.
